### PR TITLE
[draft] Add support for OSX

### PR DESCRIPTION
**Pull request by iraizo**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,8 +19,7 @@ endef
 
 CPPFLAGS := $(CPPFLAGS) -D_GNU_SOURCE -I include
 SHARED_FLAGS := -O3 -flto -fPIC -fvisibility=hidden -fno-plt \
-    $(call safe_flag,-fstack-clash-protection) -fstack-protector-strong -pipe -Wall -Wextra \
```
Comment (Zanthed) [inline Makefile:22]: Someone will need to look into why safe_flag doesn't work as the Makefile definition is supposed to check if the compiler supports -fstack-clash-protection. arm64 does not support -fstack-clash-protection yet. https://github.com/GrapheneOS/hardened_malloc/issues/149#issuecomment-1010526647

Also, the other arguments _should_ work. I know for a fact -pipe does.

File: Makefile [hunk 2]
```diff
@@ -19,8 +19,7 @@ endef
...
-    $(call safe_flag,-Wcast-align=strict,-Wcast-align) -Wcast-qual -Wwrite-strings
```
Comment (Zanthed) [inline Makefile:23]: Same issue here regarding safe_flag stuff.

-Wcast-qual and -Wwrite-strings definitely do work though.

File: Makefile [hunk 3]
```diff
@@ -32,15 +31,16 @@ endif
 
 CFLAGS := $(CFLAGS) -std=c11 $(SHARED_FLAGS) -Wmissing-prototypes
 CXXFLAGS := $(CXXFLAGS) -std=c++17 $(SHARED_FLAGS)
-LDFLAGS := $(LDFLAGS) -Wl,--as-needed,-z,defs,-z,relro,-z,now,-z,nodlopen,-z,text
```
Comment (Zanthed) [inline Makefile:35]: There _has_ to be an "Apple Clang" alternative to `-z,arg` here or whatever. Might wanna look into that.

**Comment by thestinger** [comment]
This is going to need substantially more work in order to support macOS. It will need to be ported to using 16k pages which is going to be quite involved. I don't want to merge anything related to it until it's completed.

**Pull request closed without merging**